code/game/machinery/accounting.dm

```diff
@@ -0,0 +1,63 @@
+/obj/machinery/accounting
+	name = "account registration device"
+	desc = "A machine that allows heads of staff to create a new bank account after inserting an ID."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "recharger"
+	circuit = /obj/item/circuitboard/machine/accounting
+	pass_flags = PASSTABLE
+	req_one_access = list(ACCESS_HEADS, ACCESS_CHANGE_IDS)
+	var/obj/item/card/id/inserted_id
```

Comment:
clean this variable both on `Destroy()` and also register to it getting deleted to clear it